In [6]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression

df = pd.read_excel("normalized_uber_data_enhanced (1).xlsx")

df = pd.get_dummies(df, columns=["Time_Slot", "Day_of_Week"], drop_first=True)

X = df.drop(columns=["Hour", "Date", "Time"])
y = df["Hour"]

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)

model = LinearRegression()
model.fit(X_train, y_train)

print("Model trained successfully")

Model trained successfully


In [7]:
y_pred = model.predict(X_test)

In [8]:
from sklearn.metrics import r2_score

print("R2 Score:", r2_score(y_test, y_pred))

R2 Score: 0.9146880512894132


Conclusion:The model uses time slot and day information to predict the hour. Since the accuracy is high (R² = 0.91), it shows that these time-related features are strongly connected to ride timing.